packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -662,24 +662,30 @@ export class PartialSequenceLengths {
 			removalInfo &&
 			(!moveInfo ||
 				moveIsLocal ||
-				(!removalIsLocal && moveInfo.movedSeq > removalInfo.removedSeq));
+				(!removalIsLocal &&
+					moveInfo.movedSeq !== undefined &&
```

Comment:
i'd lean toward adding null assertions here as well to avoid behavioral changes, but it's a nit (I think either way is correct)